SpatialGDK/Source/SpatialGDKServices/Public/LocalReceptionistProxyServerManager.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogLocalReceptionistProxyServerManager, Log, All);
+
+class FLocalReceptionistProxyServerManager
+{
+public:
+	FLocalReceptionistProxyServerManager();
+
+	bool CheckIfPortIsBound(int32 Port, FString& PID, FText& LogMessages);
+	bool TryKillBlockingPortProcess(const FString& PID);
+	bool LocalReceptionistProxyServerPreRunChecks(int32 ReceptionistPort);
+
+	bool GetProcessName(const FString& PID, FString& ProcessName);
+
+	void SPATIALGDKSERVICES_API Init(int32 ReceptionistPort);
+	bool SPATIALGDKSERVICES_API TryStartReceptionistProxyServer(bool bIsRunningInChina, const FString& CloudDeploymentName, const FString& ListeningAddress, int32 ReceptionistPort);
+	bool SPATIALGDKSERVICES_API TryStopReceptionistProxyServer();
+	
+private:
+
+	FProcHandle ProxyServerProcHandle;
+	FString RunningCloudDeploymentName;
+
+	static const int32 ExitCodeSuccess = 0;
```

Comment:
Maybe a DEFINE would be nice instead?

Reply:
Is not DEFINE more suitable for Macros? I do not think it is the best option in this case.

Reply:
I meant using #DEGINE to create a constant:

#DEGINE EXIT_CODE_SUCCESS 0

but yeah, it's not an important detail

Reply:
A constant is definitely better than a define here, but I would suggest it is moved to an anonymous namespace in the `.cpp` instead of being here in the header (because it's private so it doesn't serve any purpose in the header).